pkg/tsdb/v3iotsdb.go

```diff
@@ -182,15 +182,13 @@ func (a *V3ioAdapter) Querier(_ context.Context, mint, maxt int64) (*querier.V3i
 func (a *V3ioAdapter) DeleteDB(configExists bool, force bool, fromTime int64, toTime int64) error {
 	partitions := a.partitionMngr.PartsForRange(fromTime, toTime)
 	for _, part := range partitions {
-		if toTime == 0 || part.GetEndTime() < toTime {
-			a.logger.Info("Delete partition %s", part.GetTablePath())
-			err := utils.DeleteTable(a.logger, a.container, part.GetTablePath(), "", a.cfg.QryWorkers)
-			if err != nil && !force {
-				return err
-			}
-			// delete the Directory object
-			a.container.Sync.DeleteObject(&v3io.DeleteObjectInput{Path: part.GetTablePath()})
+		a.logger.Info("Delete partition %s", part.GetTablePath())
+		err := utils.DeleteTable(a.logger, a.container, part.GetTablePath(), "", a.cfg.QryWorkers)
+		if err != nil && !force {
+			return err
 		}
+		// delete the Directory object
+		a.container.Sync.DeleteObject(&v3io.DeleteObjectInput{Path: part.GetTablePath()})
 	}
 	path := a.cfg.Path + "/names/"
 	a.logger.Info("Delete metric names in path %s", path)
```

Comment:
Deleting all the names in case of partial delete of the database (delete obsolete partitions) seems incorrect.